harness/tests/determined/common/experimental/test_experiment.py

```diff
@@ -209,3 +210,38 @@ def test_unarchive_doesnt_update_local_on_rest_failure(
         raise AssertionError("bindings API call should raise an exception")
     except bindings.APIHttpError:
         assert expref.archived is None
+
+
+@responses.activate
+@mock.patch("determined.common.api.bindings.put_PutExperimentLabel")
+def test_add_label_url_encodes_label(
+    mock_bindings: mock.MagicMock,
+    make_expref: Callable[[int], experiment.Experiment],
+) -> None:
+    expref = make_expref(1)
+    label = "label with @#$%^&* symbols"
+    url_encoded_label = "label%20with%20%40%23%24%25%5E%26%2A%20symbols"
+    assert parse.unquote(url_encoded_label) == label
+
+    expref.add_label(label=label)
+
+    assert mock_bindings.call_count == 1
+    _, call_kwargs = mock_bindings.call_args_list[0]
+    assert call_kwargs["label"] == url_encoded_label
+
+
+@mock.patch("determined.common.api.bindings.delete_DeleteExperimentLabel")
```

Comment:
I don't remember what you decided, re: adding tests that labels don't get modified if there's a network failure. We talked about it... right? I'm good either way, just confused that I don't see a reference to it in this CR.

Reply:
unless i'm misremembering, we talked about that, but for a different feature (adding archive/unarchive on experiment), and those tests are right above this one (test_unarchive_doesnt_update_local_on_rest_failure)